File: dmn-test-cases/standard/tck/1.1/cl3/0031-static-user-defined-functions/translator/expected/java/dmn/type/TFnInvocationResult.java
package type;

import java.util.*;

@javax.annotation.Generated(value = {"itemDefinitionInterface.ftl", "tFnInvocationResult"})
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(as = type.TFnInvocationResultImpl.class)
public interface TFnInvocationResult extends com.gs.dmn.runtime.DMNType {
    static TFnInvocationResult toTFnInvocationResult(Object other) {
        if (other == null) {
            return null;
        } else if (TFnInvocationResult.class.isAssignableFrom(other.getClass())) {
            return (TFnInvocationResult)other;
        } else if (other instanceof com.gs.dmn.runtime.Context) {
            TFnInvocationResultImpl result_ = new TFnInvocationResultImpl();
            Object sumResult = ((com.gs.dmn.runtime.Context)other).get("sumResult");
            result_.setSumResult((java.lang.Number)sumResult);
            Object multiplicationResultPositional = ((com.gs.dmn.runtime.Context)other).get("multiplicationResultPositional");
            result_.setMultiplicationResultPositional((java.lang.Number)multiplicationResultPositional);
            Object divisionResultPositional = ((com.gs.dmn.runtime.Context)other).get("divisionResultPositional");
            result_.setDivisionResultPositional((java.lang.Number)divisionResultPositional);
            return result_;
        } else if (other instanceof com.gs.dmn.runtime.DMNType) {
            return toTFnInvocationResult(((com.gs.dmn.runtime.DMNType)other).toContext());
        } else {
            throw new com.gs.dmn.runtime.DMNRuntimeException(String.format("Cannot convert '%s' to '%s'", other.getClass().getSimpleName(), TFnInvocationResult.class.getSimpleName()));
        }
    }

    @com.fasterxml.jackson.annotation.JsonGetter("sumResult")
    java.lang.Number getSumResult();

    @com.fasterxml.jackson.annotation.JsonGetter("multiplicationResultPositional")
    java.lang.Number getMultiplicationResultPositional();

    @com.fasterxml.jackson.annotation.JsonGetter("divisionResultPositional")
    java.lang.Number getDivisionResultPositional();

    default com.gs.dmn.runtime.Context toContext() {
        com.gs.dmn.runtime.Context context = new com.gs.dmn.runtime.Context();
        context.put("sumResult", getSumResult());
        context.put("multiplicationResultPositional", getMultiplicationResultPositional());
        context.put("divisionResultPositional", getDivisionResultPositional());
        return context;
    }

    default boolean equalTo(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        TFnInvocationResult other = (TFnInvocationResult) o;
        if (this.getDivisionResultPositional() != null ? !this.getDivisionResultPositional().equals(other.getDivisionResultPositional()) : other.getDivisionResultPositional() != null) return false;
        if (this.getMultiplicationResultPositional() != null ? !this.getMultiplicationResultPositional().equals(other.getMultiplicationResultPositional()) : other.getMultiplicationResultPositional() != null) return false;
        if (this.getSumResult() != null ? !this.getSumResult().equals(other.getSumResult()) : other.getSumResult() != null) return false;

        return true;
    }

    default int hash() {
        int result = 0;
        result = 31 * result + (this.getDivisionResultPositional() != null ? this.getDivisionResultPositional().hashCode() : 0);
        result = 31 * result + (this.getMultiplicationResultPositional() != null ? this.getMultiplicationResultPositional().hashCode() : 0);
        result = 31 * result + (this.getSumResult() != null ? this.getSumResult().hashCode() : 0);
        return result;
    }

    default String asString() {
        StringBuilder result_ = new StringBuilder("{");
        result_.append("divisionResultPositional=" + getDivisionResultPositional());
        result_.append(", multiplicationResultPositional=" + getMultiplicationResultPositional());
        result_.append(", sumResult=" + getSumResult());
        result_.append("}");
        return result_.toString();
    }
}

File: dmn-test-cases/standard/tck/1.1/cl3/0031-static-user-defined-functions/translator/expected/kotlin/dmn/type/TFnInvocationResult.kt
package type

import java.util.*

@javax.annotation.Generated(value = ["itemDefinitionInterface.ftl", "tFnInvocationResult"])
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(`as` = type.TFnInvocationResultImpl::class)
interface TFnInvocationResult : com.gs.dmn.runtime.DMNType {
    @get:com.fasterxml.jackson.annotation.JsonGetter("sumResult")
    val sumResult: kotlin.Number?

    @get:com.fasterxml.jackson.annotation.JsonGetter("multiplicationResultPositional")
    val multiplicationResultPositional: kotlin.Number?

    @get:com.fasterxml.jackson.annotation.JsonGetter("divisionResultPositional")
    val divisionResultPositional: kotlin.Number?

    override fun toContext(): com.gs.dmn.runtime.Context {
        val context = com.gs.dmn.runtime.Context()
        context.add("sumResult", this.sumResult)
        context.add("multiplicationResultPositional", this.multiplicationResultPositional)
        context.add("divisionResultPositional", this.divisionResultPositional)
        return context
    }

    fun equalTo(o: Any?): Boolean {
        if (this === o) return true
        if (javaClass != o?.javaClass) return false

        val other = o as TFnInvocationResult
        if (if (this.divisionResultPositional != null) this.divisionResultPositional != other.divisionResultPositional else other.divisionResultPositional != null) return false
        if (if (this.multiplicationResultPositional != null) this.multiplicationResultPositional != other.multiplicationResultPositional else other.multiplicationResultPositional != null) return false
        if (if (this.sumResult != null) this.sumResult != other.sumResult else other.sumResult != null) return false

        return true
    }

    fun hash(): Int {
        var result = 0
        result = 31 * result + (if (this.divisionResultPositional != null) this.divisionResultPositional.hashCode() else 0)
        result = 31 * result + (if (this.multiplicationResultPositional != null) this.multiplicationResultPositional.hashCode() else 0)
        result = 31 * result + (if (this.sumResult != null) this.sumResult.hashCode() else 0)
        return result
    }

    fun asString(): String {
        val result_ = StringBuilder("{")
        result_.append("divisionResultPositional=" + divisionResultPositional)
        result_.append(", multiplicationResultPositional=" + multiplicationResultPositional)
        result_.append(", sumResult=" + sumResult)
        result_.append("}")
        return result_.toString()
    }

    companion object {
        @JvmStatic
        fun toTFnInvocationResult(other: Any?): TFnInvocationResult? {
            if (other == null) {
                return null
            } else if (other is TFnInvocationResult?) {
                return other
            } else if (other is com.gs.dmn.runtime.Context) {
                var result_ = TFnInvocationResultImpl()
                var sumResult = other.get("sumResult")
                result_.sumResult = sumResult as kotlin.Number?
                var multiplicationResultPositional = other.get("multiplicationResultPositional")
                result_.multiplicationResultPositional = multiplicationResultPositional as kotlin.Number?
                var divisionResultPositional = other.get("divisionResultPositional")
                result_.divisionResultPositional = divisionResultPositional as kotlin.Number?
                return result_
            } else if (other is com.gs.dmn.runtime.DMNType) {
                return toTFnInvocationResult(other.toContext())
            } else {
                throw com.gs.dmn.runtime.DMNRuntimeException(String.format("Cannot convert '%s' to '%s'", other.javaClass.getSimpleName(), TFnInvocationResult::class.java.getSimpleName()))
            }
        }
    }
}
